### PR TITLE
Add powershell script for publishing OpenRCT2

### PR DESCRIPTION
**Pull request by IntelOrca**
In order to achieve #2406 and for the benefit of other build servers for Windows, this PR adds a publish powershell script to update the #defines in rct2.h, build the project and create a zip.

To be consistent with the bash equivalent, I am thinking of renaming pre-build.ps1 to install.ps1 and making so you have to always manually run it rather than it being included in the msbuild project.

Improvements are welcome into making the scripts more robust. One of which could be to check that git, msbuild and 7za are in path or otherwise try to find it.

### CODE REVIEW

**Comment by janisozaur** [comment]
Oh cool, just today I ran into someone still using XP so I did similar thing in my appveyor config.

Since there is also `Release XP` profile, could you possibly add that as well to build and package steps?

**Comment by IntelOrca** [comment]
@janisozaur didn't know it could do that - thanks

**Comment by janisozaur** [comment]
regarding how you set defines:
I mostly encountered it being done this way:
```
#ifndef PROJECT_VERSION
#define PROJECT_VERSION ""
#endif
```
Then fetch the version while invoking build command and setting preprocessor defines, like
```
gcc -DPROJECT_VERSION=12345 …
```

The second option is to generate `version.h` with all the defines in it, without any checks what's in the file.

I don't think I have ever seen a script trying to parse C source and modify it on the fly. I'd say that's the most complex and error-prone method out of these three.

Personally I like the first one best, how do you feel about switching to it?

**Comment by IntelOrca** [comment]
@janisozaur I hadn't really thought of it like that. So yes, it could just be switched to adding them as directives to msbuild

**Comment by janisozaur** [comment]
And don't forget to bump `appveyor.yml` to `0.0.4` now

File: scripts/build.ps1 [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    msbuild $rootPath\openrct2.sln /p:Configuration=$Configuration /p:Platform=Win32 $target /v:minimal | Write-Host
+    return $LASTEXITCODE
+}
+
+function Build-G2()
+{
+    # Check if OpenRCT2 binary exists
+    if (-not (Test-Path $openrct2Path))
+    {
+        Write-Host "You must build OpenRCT2 first before g2.dat can be built." -ForegroundColor Red
```
Comment (janisozaur) [inline scripts/build.ps1]: With CMake (as it is common in Linux projects), when a target requires some other target, we simply set the dependency chain and let `make` worry about that dependency being resolved. In other words: on Linux I would expect `make g2` to figure out it depends on target `openrct2`, build it if not current, then proceed to making target `g2`.

Is this approach (to bail out) common on Windows?

File: scripts/build.ps1 [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+# Script entry point
+switch ($Task)
+{
+    "data"     { $result = Build-Data }
+    "openrct2" { $result = Build-OpenRCT2 }
+    "g2"       { $result = Build-G2 }
+    "all"      { $result = Build-All }
+    default    { Write-Host "Unknown build task." -ForegroundColor Red
```
Comment (janisozaur) [inline scripts/build.ps1]: `make` implies `make all`, which usually is set to the most likely target, with CMake we have it set to `openrct2`.

Is it common on Windows to bail out if no explicit target is given?

**Comment by PFCKrutonium** [comment]
>To be consistent with the bash equivalent, I am thinking of renaming pre-build.ps1 to install.ps1 and making so you have to always manually run it rather than it being included in the msbuild project.

This breaks the build server :/

**Pull request closed without merging**